util.go

```diff
@@ -0,0 +1,54 @@
+package main
```

Comment:
please confirm if this is all copy pasted code and nothing is modified

util.go and util_test.go

Reply:
Yes this is just copy pasted.